### ISSUE TITLE
Incorrect parsing of `\r`-delimited lines

### ISSUE THREAD

**Issue by revin** [markdown-it/markdown-it#252]
Hello folks, thanks as always for the wonderful library! We ran into a bug with the parsing of a file with `\r` line endings:

```js
var md = new require('markdown-it')();
md.render('# Heading\rThis is a thing\r\r* item one\r* item two');
```

renders as
> `'<h1>Foo\rThis is a thing\r\r- item one\r- item two</h1>\n'`

rather than the expected
> `'<h1>Heading</h1>\n<p>This is a thing</p>\n<ul>\n<li>item one\nitem two</li>\n</ul>\n'`

I guess because [`NEWLINES_RE`](https://github.com/markdown-it/markdown-it/blob/master/lib/rules_core/normalize.js#L6) isn't matching the bare `\r` characters? What do you think? Thanks in advance! :+1:

**Comment by puzrin** [comment]
I'm not sure if plain `\r` without `\n` is valid or not. My suggestion to all such questions is the same - check reference implementation:

- https://github.com/jgm/commonmark.js.
- in browser (not for this case) http://spec.commonmark.org/dingus/

If you found a deviation from reference, it will be fixed immediately. If reference has the same behaviour - then question should be directed to CM tracker/forum or patched on your side.

**Comment by wooorm** [comment]
I think markdown-it deviates from CM here ([spec](http://spec.commonmark.org/0.25/#line-ending)):

> A line ending is a newline (U+000A), **a carriage return (U+000D) not followed by a newline**, or a carriage return and a following newline.

**Comment by puzrin** [comment]
@wooorm thanks for clear point.

@rlidwka please check & fix.

**Comment by rlidwka** [comment]
Fixed, thanks!

**Comment by revin** [comment]
That was quick. Thanks everyone!

**Comment by puzrin** [comment]
Released 6.0.5 with this fix. Enjoy!

PS. @revin I also recommend to update or at least reinstall `markdown-it` with fresh dependencies at npm registry service. You need updated `linkify-it` with fixed ReDOS. See [6.0.3](https://github.com/markdown-it/markdown-it/blob/master/CHANGELOG.md#603--2016-05-30) changelog.
